src/components/charts/SwipeChart.tsx

```diff
@@ -215,33 +213,33 @@ const SwipeChartComponent = (params: Props) => {
             // Construct the dataset query
             const response = await fetch(fetchUrl)
             const result = await response.json()
-            const marketChartRange = asCoinGeckoMarketApi(result)
-            if ('status' in marketChartRange) {
-              if (marketChartRange.status.error_code === 429) {
+            const apiError = asMaybe(asCoinGeckoError)(result)
+            if (apiError != null) {
+              if (apiError.status.error_code === 429) {
                 // Rate limit error, use our API key as a fallback
-                if (!fetchUrl.includes('x_cg_pro_api_key')) {
+                if (!fetchUrl.includes('x_cg_pro_api_key') && ENV.COINGECKO_API_KEY !== '') {
                   fetchUrl = `${COINGECKO_URL_PRO}${fetchPath}&x_cg_pro_api_key=${ENV.COINGECKO_API_KEY}`
                   continue
```

Comment:
The continue should be moved outside of the apikey block so that rate-limited requests without keys can be retried as well.

Reply:
The continue retries the request. Why should it be outside of the retry path?

Reply:
It needs to be in the 429 block but outside of the apikey block